tools/read-version

```diff
@@ -65,7 +65,13 @@ output_json = '--json' in sys.argv
 src_version = ci_version.version_string()
 version_long = None
 
-if is_gitdir(_tdir) and which("git"):
+# If we're performing CI for a new release branch (which our tooling creates
+# with an "upstream/" prefix), then we don't want to enforce strict version
+# matching because we know it will fail.
+is_release_branch_ci = (
+    os.environ.get("TRAVIS_PULL_REQUEST_BRANCH", "").startswith("upstream/")
```

Comment:
Why do we need this startswith check?
Shoudn't we be able to check bool(os.environ.get("TRAVIS_PULL_REQUEST_BRANCH"))  as TRAVIS_PULL_REQUEST_BRANCH should be "" on push builds?
https://docs.travis-ci.com/user/environment-variables/

Reply:
We don't expect this check to fail in anything but branches that are introducing a new upstream release (which our tooling names `upstream/...`). This check is scoped as tightly to that scenario as possible. If this check fails on any other pull request branch, it's a legit failure that needs fixing.